eng/pipelines/templates/jobs/archetype-sdk-tests-generate.yml

```diff
@@ -20,23 +23,31 @@ jobs:
     - ${{ each config in parameters.MatrixConfigs }}:
       - ${{ if eq(config.GenerateVMJobs, 'true') }}:
         - pwsh: |
-            Write-Host "DISPLAY NAME FILTER: $(displayNameFilter)"
+            $extraFilters = "${{ join('","', parameters.MatrixFilters) }}"
+            $filters = ,"container=^$","SupportedClouds=^$|${{ parameters.CloudConfig.Cloud }}"
+            if ($extraFilters) { $filters = @($extraFilters) + $filters }
+            Write-Host "DISPLAY NAME FILTER: `"$(displayNameFilter)`"; FILTERS: `"$filters`""
             eng/scripts/job-matrix/Create-JobMatrix.ps1 `
```

Comment:
One thing to think about from a debugging standpoint is how someone would run this script locally to see the results. We generally use the [Powershell task](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/2/Azure-DevOps-Pipeline-Guidance?anchor=devops-tasks-vs-pwsh-script-shortcut) directly because it also outputs the command line that you could copy and paste. Given your pre-processing of the args inline that might be difficult but we should see if we can come up with a way to show the full command line so that we can more easily repro locally if needed.

Reply:
I guess one option would be to also enable debug tracing (https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/set-psdebug?view=powershell-7.1#example-1--set-the-trace-level) which would have each line in this pwsh inline script echoed before executing.

Reply:
For filters, it will print them out in the log. Typically what I do to see the arguments is run the [validation script](https://dev.azure.com/azure-sdk/internal/_git/azure-sdk-devops?path=%2Fvalidate-pipelineyaml.ps1) which outputs the generated script to disk, though the devops variable substitutions will still be there. Then you can copy paste and run, e.g.

```
          $extraFilters = ""
          $filters = ,"container=^$","SupportedClouds=^$|Public"
          if ($extraFilters) { $filters = @($extraFilters) + $filters }
          Write-Host "DISPLAY NAME FILTER: `"$(displayNameFilter)`"; FILTERS: `"$filters`""

          eng/scripts/job-matrix/Create-JobMatrix.ps1 `
            -ConfigPath sdk/tables/platform-matrix.json `
            -Selection sparse `
            -DisplayNameFilter "$(displayNameFilter)" `
            -Filters $filters `
            -NonSparseParameters "JavaTestVersion","AZURE_TEST_HTTP_CLIENTS"
```

Reply:
The reason I'm not using the powershell task directly (as I remember you suggesting in other cases), is because, oddly enough, it doesn't support the `name` field, only displayName! In this case, we need to set the name field dynamically, since it's in a loop, and we need to reference the job's variable outputs by the job name:

```
name: generate_vm_job_matrix_${{ config.Name }}
...
Matrix: dependencies.generate_matrix.outputs['generate_vm_job_matrix_${{ config.Name }}.matrix']
```

Reply:
We could print out all argument values from the Create-JobMatrix script? (or even format the log so it prints a usable command to test).

Reply:
name and display name are on the common parameter set of all tasks so I'd expect those to be present for the Powershell task, however they might not be if you use the pwsh shorthand. 

At any rate it might still be worth using my Set-PSDebug suggestion so we can easily have something to copy/paste to run on locally.

Reply:
Yeah I think I was looking at the task inputs schema, and not realizing there was a higher level one to see. Testing this out.